Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1386 @@
+        /// </summary>
+        public override void Dispose()
+        {
+        }
+
+        private void OnError(object sender, WebSocketError e)
+        {
+            Log.Error($"ZerodhaSubscriptionManager.OnError(): Message: {e.Message} Exception: {e.Exception}");
+        }
+
+        private void OnMessage(object sender, MessageData e)
+        {
+            LastHeartbeatUtcTime = DateTime.UtcNow;
+
+            // Verify if we're allowed to handle the streaming packet yet; while we're placing an order we delay the
+            // stream processing a touch.
+            try
+            {
+                if (_streamLocked)
+                {
+                    _messageBuffer.Enqueue(e);
+                    return;
+                }
+            }
+            catch (Exception err)
+            {
+                Log.Error(err);
+            }
+
+            OnMessageImpl(e);
+        }
+
+        /// <summary>
+        /// Implementation of the OnMessage event
+        /// </summary>
+        /// <param name="e"></param>
+        private void OnMessageImpl(MessageData e)
+        {
+            try
+            {
+                if (e.MessageType == WebSocketMessageType.Binary)
+                {
+                    if (e.Count > 1)
+                    {
+                        int offset = 0;
+                        ushort count = ReadShort(e.Data, ref offset); //number of packets
+
+                        for (ushort i = 0; i < count; i++)
+                        {
+                            ushort length = ReadShort(e.Data, ref offset); // length of the packet
+                            Messages.Tick tick = new Messages.Tick();
+                            if (length == 8) // ltp
+                                tick = ReadLTP(e.Data, ref offset);
+                            else if (length == 28) // index quote
+                                tick = ReadIndexQuote(e.Data, ref offset);
+                            else if (length == 32) // index quote
+                                tick = ReadIndexFull(e.Data, ref offset);
+                            else if (length == 44) // quote
+                                tick = ReadQuote(e.Data, ref offset);
+                            else if (length == 184) // full with marketdepth and timestamp
+                                tick = ReadFull(e.Data, ref offset);
+                            // If the number of bytes got from stream is less that that is required
+                            // data is invalid. This will skip that wrong tick
+                            if (tick.InstrumentToken != 0 && offset <= e.Count && tick.Mode == Constants.MODE_FULL)
+                            {
+                                var symbol = _symbolMapper.ConvertZerodhaSymbolToLeanSymbol(tick.InstrumentToken);
+
+                                EmitQuoteTick(symbol, tick.Bids, tick.BuyQuantity, tick.Offers, tick.SellQuantity, tick.Timestamp.GetValueOrDefault());
```

Comment:
Is `tick.Timestamp` in UTC? If yes, it needs to be converted to exchange time zone

Same for `EmitTradeTick`